controllers/addresspool_controller.go

```diff
@@ -80,8 +80,8 @@ func renderObject(instance *metallbv1alpha1.AddressPool) ([]*unstructured.Unstru
 	data.Data["AutoAssign"] = *instance.Spec.AutoAssign
 	data.Data["Addresses"] = instance.Spec.Addresses
 	objs, err := render.RenderDir(AddressPoolManifestPath, &data)
-	if err != nil {
-		return nil, fmt.Errorf("Fail to render address-pool manifest %v", err)
+	if err != nil || objs == nil {
```

Comment:
why is this needed?

Reply:
I wanted to avoid doing len of nil in

Reply:
Yes but objs == nil will never be true, because worst case objs is what `out` is being initialized here https://github.com/metallb/metallb-operator/blob/0c5b52808ce215e85b79e9cf0069e5cb4ebdad99/pkg/render/render.go#L34